test/loading.jl

```diff
@@ -77,6 +77,10 @@ mktempdir() do dir
             @test Base.isfile_casesensitive(true_filename)
             @test !Base.isfile_casesensitive(lowered_filename)
 
+            # check that case-sensitivity is preserved for relative paths
+            @test Base.isfile_casesensitive(joinpath("dummy", "..", true_filename))
+            @test !Base.isfile_casesensitive(joinpath("dummy", "..", lowered_filename))
```

Comment:
```suggestion
            @test Base.isfile_casesensitive(joinpath("..", "test", true_filename))
            @test !Base.isfile_casesensitive(joinpath("..", "test", lowered_filename))
```
use an actual path component, not a dummy component?